app-frontend/src/app/components/diagramContainer/diagramContainer.controller.js

```diff
@@ -17,82 +20,177 @@ export default class DiagramContainerController {
             this.graph.clear();
             this.initShapes();
             this.shapes.forEach(s => this.graph.addCell(s));
+            this.initDiagram();
         }
     }
 
     initShapes() {
         this.shapes = [];
-        this.shapes.push(new joint.shapes.basic.Rect({
-            position: {
-                x: ($(this.workspaceElement).width() - 225) / 2,
-                y: ($(this.workspaceElement).height() - 75) / 2
-            },
+
+        this.createRectangle('NDVI 1', {
+            label: 'NDVI',
+            inputs: ['Red', 'NIR'],
+            outputs: ['Output']
+        });
+
+        this.createRectangle('Reclassify 1', {
+            label: 'Reclassify',
+            inputs: ['Input'],
+            outputs: ['Output']
+        });
+
+        this.createLink(['NDVI 1', 'Output'], ['Reclassify 1', 'Input']);
+
+        this.createRectangle('NDVI 2', {
+            label: 'NDVI',
```

Comment:
I think maybe a better name here between both NDVI rectangles would be nice, just so we can match up which is before an event and which is after